docs_src/design/adr/device-service/0012-DeviceService-Filters.md

```diff
@@ -0,0 +1,151 @@
+
+After receiving the `CommandValues` the `execReadDeviceResource` function calls the `cvsToEvent` function convert the `CommandValues` into and `Event/Reading` objects (from `go-mod-core-contracts`) and returns this model to be sent via REST to core data by the rest of the SDK.
+
+It is precisely after the convert to `Event/Reading` objects (after the async readings are assembled into events) and before returning that result in `common.SendEvent` (in utils.go) function that the device service should invoke the required filter functions.  In the existing V1 implementation of the device-sdk-go, commands, async readings, and auto-events all call the function `common.SendEvent()`.  This means that there would likely be several different places in the code which would need to call `FilterEvent`.  In the C SDK, it is likely that the filters will be called before conversion to Event/Reading objects - they will operate on commandresult objects (equivalent to CommandValues).
+
+The order in which functions are called is important when more than one filter is provided.  The order that functions are called should be reflected in the order listed in the configuration of the filters.
+
+Events containing binary values (event.HasBinaryValue), will not be filtered.  Future releases may include binary value filters.
+
+#### Setting Filter Function and Configuration
+
+When filter functions are shared (or appear to be doing the same type of work) between SDKs, the configuration of the similar filter functions should also look similar.  The app functions SDK configuration model for filters should therefore be followed.
+
+While device services do not have pipelines, the inclusion and configuration of filters for device services should have a similar look (to provide symmetry with app services). The configuration has to provide the functions required and parameters to make the functions work - even though the association to a pipeline is not required.  Below is the common app service configuration as it relates to filters:
+
+``` toml
+[Writable.Pipeline]
+    ExecutionOrder = "FilterByDeviceName, TransformToXML, SetOutputData"
+    [Writable.Pipeline.Functions.FilterByDeviceName]
+    [Writable.Pipeline.Functions.FilterByDeviceName.Parameters]
+        DeviceNames = "Random-Float-Device,Random-Integer-Device"
+        FilterOut = "false"
+```
+
+Suggested and hypothetical configuration for the device service reading filters should look something like that below.
+
+``` toml
+[Writable.Filters]
+    ExecutionOrder = "FilterByValueInRange, FilterByResourceNamesMatch"
+    [Writable.Filter.Functions.FilterByResourceNamesMatch]
+        [Writable.Filter.Functions.FilterByResourceNamesMatch.Parameters]
+            DeviceNames = "Random-Float-Device,Random-Integer-Device"
+            FilterOut = "false"
+    [Writable.Filter.Functions.FilterByValueInRange]
+        [Writable.Filter.Functions.FilterByValueInRange.Parameters]
```

Comment:
@tonyespy has a good point about this range filtering needing to be specific to a Resource. So need to add ResourceName as a parameter.